MRUBY_CONFIG not globally available as before

Hi @shuujii ,

I'm trying to compile [mruby-esp32](https://github.com/mruby-esp32/mruby-esp32) examples.
It compiles ok against 2.1.2 mruby. But not against master branch.
It issues a `NameError: uninitialized constant #<Class:MRuby::Lockfile>::MRUBY_CONFIG`

I think this error was introduced at 81c21088196fd92d08c253efc2feb8dbead88acf
Before that `MRUBY_CONFIG` was being set in Rakefile and readly available.
But from this patch on `MRUBY_CONFIG` is being set "after" running `MRuby::Build.load_config`.
The problem rises because `MRuby::Lockfile.instance` is called before `MRUBY_CONFIG` is set.
I'm opening a PR for discussion.

The complete error message follows.
```
make MRUBY_EXAMPLE=wifi_example_mrb.rb flash monitor
Toolchain path: /Users/abinoam/.espressif/tools/xtensa-esp32-elf/esp-2020r3-8.4.0/xtensa-esp32-elf/bin/xtensa-esp32-elf-gcc
Toolchain version: esp-2020r3
Compiler version: 8.4.0
Python requirements from /Users/abinoam/esp/esp-idf/requirements.txt are satisfied.

App "mruby_example" version: a08fb71-dirty
rake aborted!
NameError: uninitialized constant #<Class:MRuby::Lockfile>::MRUBY_CONFIG
/Users/abinoam/esp/mruby-esp32/components/mruby_component/mruby/lib/mruby/lockfile.rb:29:in `instance'
/Users/abinoam/esp/mruby-esp32/components/mruby_component/mruby/lib/mruby/lockfile.rb:21:in `build'
/Users/abinoam/esp/mruby-esp32/components/mruby_component/mruby/lib/mruby/build.rb:253:in `locks'
/Users/abinoam/esp/mruby-esp32/components/mruby_component/mruby/lib/mruby/build/load_gems.rb:87:in `load_special_path_gem'
/Users/abinoam/esp/mruby-esp32/components/mruby_component/mruby/lib/mruby/build/load_gems.rb:17:in `gem'
/Users/abinoam/esp/mruby-esp32/components/mruby_component/esp32_build_config.rb:61:in `block in <top (required)>'
/Users/abinoam/esp/mruby-esp32/components/mruby_component/mruby/lib/mruby/build.rb:111:in `instance_eval'
/Users/abinoam/esp/mruby-esp32/components/mruby_component/mruby/lib/mruby/build.rb:111:in `initialize'
/Users/abinoam/esp/mruby-esp32/components/mruby_component/mruby/lib/mruby/build.rb:400:in `initialize'
/Users/abinoam/esp/mruby-esp32/components/mruby_component/esp32_build_config.rb:28:in `new'
/Users/abinoam/esp/mruby-esp32/components/mruby_component/esp32_build_config.rb:28:in `<top (required)>'
/Users/abinoam/esp/mruby-esp32/components/mruby_component/mruby/lib/mruby/build.rb:50:in `load'
/Users/abinoam/esp/mruby-esp32/components/mruby_component/mruby/lib/mruby/build.rb:50:in `load_config'
/Users/abinoam/esp/mruby-esp32/components/mruby_component/mruby/Rakefile:17:in `<top (required)>'
/Users/abinoam/.rvm/gems/ruby-2.6.3/gems/rake-13.0.1/exe/rake:27:in `<top (required)>'
/Users/abinoam/.rvm/gems/ruby-2.6.3/bin/ruby_executable_hooks:24:in `eval'
/Users/abinoam/.rvm/gems/ruby-2.6.3/bin/ruby_executable_hooks:24:in `<main>'
(See full trace by running task with --trace)
make[2]: *** [all] Error 1
make[1]: *** [build] Error 2
make: *** [component-mruby_component-build] Error 2
```

--- a/Rakefile
+++ b/Rakefile
@@ -14,7 +14,8 @@ require "mruby/core_ext"
 require "mruby/build"
 
 # load configuration file
-MRUBY_CONFIG = MRuby::Build.load_config
+MRUBY_CONFIG = MRuby::Build.mruby_config_path
+load MRUBY_CONFIG
 
 # load basic rules
 MRuby.each_target do |build|

--- a/lib/mruby/build.rb
+++ b/lib/mruby/build.rb
@@ -39,7 +39,7 @@ module MRuby
   class Build
     class << self
       attr_accessor :current
-      def load_config
+      def mruby_config_path
         path = ENV['MRUBY_CONFIG'] || ENV['CONFIG']
         if path.nil? || path.empty?
           path = "#{MRUBY_ROOT}/build_config/default.rb"
@@ -47,7 +47,6 @@ module MRuby
           f = "#{MRUBY_ROOT}/build_config/#{path}.rb"
           path = File.exist?(f) ? f : File.extname(path).empty? ? f : path
         end
-        load path
         path
       end
     end

